### ISSUE TITLE
Extension API provides a way to listen on Kubernetes resources events

### ISSUE THREAD

**Issue by feloy** [containers/podman-desktop#9970]
### Is your enhancement related to a problem? Please describe

As an extension developer, it is not possible to know the resources present in the current Kubernetes cluster.

### Describe the solution you'd like

I would like to be able to get the list of resources (mainly pods) in a reactive way, the same way it is possible to get events on podman's resources (containers, pods, images, etc) with the `containerEngine.onEvent` method.

**Comment by benoitf** [comment]
I'm wondering if it is part of the extension point of Podman Desktop.

Or the extension itself decide to subscribe using its own library

Kubernetes could be optional, if it's in the core API it can't be an option anymore.

**Comment by feloy** [comment]
Yes, I'm also wondering. Or now that extensions can share their own api, having the (a) Kubernetes extension providing this API?

**Comment by benoitf** [comment]
yes it is an option I thought.

But maybe we could do it in the reverse way. The extension that needs to use that, use the kubernetes-client API and get the events, etc. Then once the extension is done, we could see how to move some part to another extension or provide an API on top of it.

Like for the podman extension, first, it has groomed inside the extension for its internal usage, and then we were able to expose/export some API from it.
